src/shared/util.js

```diff
@@ -710,19 +710,27 @@ class Util {
   // The SVD can be represented with formula A = USV. We are interested in the
   // matrix S here because it represents the scale values.
   static singularValueDecompose2dScale(m) {
-    const transpose = [m[0], m[2], m[1], m[3]];
+    if (Math.abs(m[1]) <= Number.EPSILON && Math.abs(m[2]) <= Number.EPSILON) {
+      // Diagonal matrix
+      return [Math.abs(m[0]) || 1, Math.abs(m[3]) || 1];
+    }
+
+    if (Math.abs(m[0]) <= Number.EPSILON && Math.abs(m[3]) <= Number.EPSILON) {
+      // Anti-diagonal matrix: scale composed with rotation of +/- 90°
+      return [Math.abs(m[2]) || 1, Math.abs(m[1]) || 1];
+    }
```

Comment:
Given the additional checks added here, how much of a performance increase are these changes bringing in the general case (do you at least have some ballpark numbers)?